test/test_functions

```diff
@@ -1181,3 +1188,125 @@ cp_usrbin() {
     sudo chmod 0444 $f || return 3
   done
 }
+
+
+# waits until cvmfs is mounted
+wait_fqrn_mount() {
+  local waits=0
+  while ! mount | grep "/cvmfs/${FQRN}" > /dev/null 2>&1; do
+    waits=$(( waits + 1 ))
+    if [ "$waits" -ge 10 ]; then
+      die "${FQRN} could not be mounted (not superuser?)"
+    fi
+    sleep 1
+  done
+}
+
+
+# runs the command once in order to get a warm cache
+create_warm_cache() {
+  local workspace=$1
+  cvmfs2 -f -o config=cvmfs.conf,disable_watchdog,simple_options_parsing $FQRN /cvmfs/$FQRN &> /dev/null &
+  wait_fqrn_mount
+
+  bash -c "cvmfs_run_benchmark" > /dev/null 2>&1
+  umount /cvmfs/$FQRN
+  return 0
+}
+
+
+# creates the default cvmfs.conf file for normal testing
+#
+# @param destination_file  absolute path of the directory
+create_default_cvmfs_config() {
+  local destination_directory="$1"
+  echo 'CVMFS_CACHE_BASE=/root/cvmfs_testing/cache
+CVMFS_RELOAD_SOCKETS=/root/cvmfs_testing/cache
+CVMFS_SERVER_URL=http://cvmfs-stratum-one.cern.ch/cvmfs/@fqrn@
+CVMFS_SHARED_CACHE=no
+CVMFS_HTTP_PROXY=http://ca-proxy.cern.ch:3128
+CVMFS_AUTO_UPDATE=false' > $destination_directory/cvmfs.conf
+}
+
+
+# sets the valgrind options depending on the test type
+#
+# @param test_type  concrete test to execute
+set_valgrind_options() {
+  local CVMFS_OPT_CALLGRIND="--tool=callgrind --dump-before=fuse_session_loop_mt --dump-after=fuse_session_loop_mt --dsymutil=yes --log-file=${FQRN}.callgrind.log -v"
+  local CVMFS_OPT_MEMCHECK="--tool=memcheck --xml=yes --xml-file=${FQRN}.memcheck.xml --trace-children=yes --child-silent-after-fork=yes --leak-check=yes --log-file=${FQRN}.memcheck.log -v"
+  OPTIONS=""
+  case $CVMFS_OPT_TEST_TYPE in
+    callgrind)
+      OPTIONS=$CVMFS_OPT_CALLGRIND
+    ;;
+    memcheck)
+      OPTIONS=$CVMFS_OPT_MEMCHECK
+    ;;
+  esac
+}
+
+
+# creates the environment to run the benchmark
+#
+# @param workspace  absolute path to the directory where the benchmark will put the files in
+setup_benchmark_environment() {
```

Comment:
I was referring to our initial finding when dealing with igprof: if /etc/mtab is not a symlink, rename to /etc/mtab.save and create a symlink /etc/mtab --> /proc/mounts.